clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4102,6 +4102,11 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     if (Args.hasFlag(options::OPT_fsycl_esimd, options::OPT_fno_sycl_esimd,
                      false))
       CmdArgs.push_back("-fsycl-explicit-simd");
+
+    if (!Args.hasFlag(options::OPT_fsycl_std_optimizations,
+                      options::OPT_fno_sycl_std_optimizations, true))
+      CmdArgs.push_back("-fno-sycl-std-optimizations");
```

Comment:
Could you add a driver test?  Otherwise, LGTM

Reply:
Done.